src/lib/Form/Data/TrashItemData.php

```diff
@@ -91,6 +91,8 @@ public function setAncestors(array $ancestors)
 
     public function isParentInTrash(): bool
     {
-        return !end($this->ancestors) instanceof Location;
+        $lastAncestor = end($this->ancestors);
+
+        return $this->location->pathString !== $lastAncestor->pathString . $this->location->id . '/';
```

Comment:
I don't think resolving this problem by string comparison is suggested resolution here. `$this->location->path` provides array of consecutive locations.  Comparing arrays is more reliable in general. cc @Nattfarinn